### ISSUE TITLE
Hidden KML features still incur hit detection performance costs (2D Mode)

### ISSUE THREAD

**Issue by jsalankey** [ngageoint/opensphere#237]
Hidden KML features still go through a significant amount of the Openlayers hit detection stack. This includes getting all the way to a costing `getImageData` call on each hidden feature that shouldn't happen.

It's exclusive to KML layers due to how we manage their shown/hidden state on an individual feature level (and we never actually disable the overall layer visibility).

**Comment by wallw-bits** [comment]
I don't think this has anything to do with overall layer visibility. What happens to hidden features on other sources?

**Comment by jsalankey** [comment]
I think when layers are invisible, they don't make it through to the portion of the hit detection doing the slow, unnecessary calls. This is likely a problem for hidden features in other sources as well, it's just more readily apparent with KML.

**Comment by schmidtk** [comment]
OpenLayers still hit detects features skipped by `ol.PluggableMap#skipFeature`. I thought I had fixed that problem with [this mixin](https://github.com/ngageoint/opensphere/blob/master/src/os/mixin/mixin.js#L59)... I'll look into it.

**Comment by schmidtk** [comment]
Ok so this seems to be an OpenLayers bug in their [skipped feature handling](https://github.com/openlayers/openlayers/blob/master/src/ol/render/canvas/Replay.js#L589). If the feature is skipped, they shouldn't handle the `END_GEOMETRY` instruction which is making the `getImageData` call. They should just move on to the next feature like they do when the feature doesn't intersect the hit extent. I'll open an issue with them.
